python/grass/temporal/register.py

```diff
@@ -31,6 +31,44 @@
 ###############################################################################
 
 
+def build_abstract_map_dataset(map_item, element):
+    """Build AbstractMapDataset from map item and element type.
```

Comment:
It build an id. The whole AbstractMapDataset here and in return doc is misleading.

But perhaps more importantly, compare this to the build_id. That's already what the function is trying to do, its doc says: _Convenient method to build the unique identifier_.

Reply:
Right. The looking at the `build_id` function does almost exactly the same:
https://github.com/OSGeo/grass/blob/1a96f69ccf430caa03bec8dbfc990c07c81f3164/python/grass/temporal/abstract_map_dataset.py#L168-L194

So, `build_id` could be extended with an _elements_ key-word that defaults to _None_, and the method would check for map presence only if:

1. an element type to look for is provided,
3. the mapset is not contained in the name, and
4. 3. no mapset is given.

That way current behavior is kept, however, the method would get an additional if check with three conditions...

Reply:
Well, the build_id function currently constructs an id from whatever it is given regardless of the existence or non-existence of the element. Adding a check for existence for a specific combination of parameters, but not the others would be hard to document and tricky to use.

get_mapset_for_map or get_mapset_for_existing_map makes more sense to me as that's more clear about what is required and what may happen even without reading the documentation (even get_mapset_for_map suggests that it must exists since only existing maps have mapsets).

It could also be something like `map, layer, mapset = ensure_mapset(name, element)`. Than the result will be passed to build_id by the caller. In any case, these would untangle the existence requirements from build_id (or its potential wrapper).

Unfortunately, the function cannot be more general than that because here we need the `:layer` piece which is not used elsewhere AFAIK.

Or maybe both your new function and build_id call a same new function to parse out the name, split_name_layer_mapset or something like that. While build_id then proceeds to create the id, your function does the existence check and ensures mapset is provided if missing.

Just to be clear, I don't see a one clear solution. Nevertheless, I think the primary goal should be to avoid creating code with unclear purpose and functions with hard-to-guess behavior.

Reply:
See below...